File: 01-Анализ и разработка алгоритмов/2024-ОЧКА-осень/fib/fib.py
memo = dict()

def fib(n):
    if n == 0: return 0
    if n == 1: return 1
    if n in memo: return memo[n]
    memo[n] = fib(n - 1) + fib(n - 2)
    return memo[n]


n = 80  # 23416728348467685
print(fib(n))

File: 01-Анализ и разработка алгоритмов/2024-ОЧКА-осень/fib/fib.js
const memo = {};

function fibonaci(n) {
    if (n === 0) return 0;
    if (n === 1) return 1;
    if (memo[n] !== undefined) return memo[n];
    memo[n] = fibonaci(n - 1) + fibonaci(n - 2);
    return memo[n];
}

let n = 80; // 23416728348467684
console.log(fibonaci(n));
